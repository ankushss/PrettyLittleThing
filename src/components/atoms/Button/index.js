import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

export const Button = ({
  children,
  onPress,
  variant = 'contained',
  ...props
}) => {
  const containedStyles = {
    backgroundColor: '#00aa5b',
    borderColor: '#00aa5b',
    borderWidth: 1,
  };

  const outlinedStyles = {
    borderColor: '#00aa5b',
    borderWidth: 1,
  };

  const containedTextStyles = {
    color: 'white',
  };

  const outlinedTextStyles = {
    color: '#00aa5b',
  };

  return (
    <TouchableOpacity onPress={onPress} {...props}>
      <View
        style={[
          styles.container,
          variant === 'contained' ? containedStyles : outlinedStyles,
        ]}>
        <Text
          style={[
            styles.text,
            variant === 'contained' ? containedTextStyles : outlinedTextStyles,
          ]}>
          {children}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
  },
  text: {
    fontWeight: 'bold',
  },
});
