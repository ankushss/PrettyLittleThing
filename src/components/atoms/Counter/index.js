import React, {useEffect, useState} from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

export const Counter = ({value = 0, onPressAdd, onPressRemove}) => {
  const [count, setCount] = useState(value);

  useEffect(() => {
    setCount(value);
  }, [value]);

  return (
    <View style={styles.container}>
      <TouchableOpacity testID="addButton" onPress={onPressAdd}>
        <View style={styles.button}>
          <Text>+</Text>
        </View>
      </TouchableOpacity>
      <Text style={styles.text}>{count}</Text>
      <TouchableOpacity testID="removeButton" onPress={onPressRemove}>
        <View style={styles.button}>
          <Text>-</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    backgroundColor: 'lightgray',
    width: 30,
    height: 30,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 100,
  },
  text: {
    marginHorizontal: 10,
    fontSize: 16,
  },
});
