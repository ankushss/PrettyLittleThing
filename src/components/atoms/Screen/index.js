import React from 'react';
import {StyleSheet, View} from 'react-native';

export const Screen = ({children, style}) => {
  return <View style={[styles.container, style]}>{children}</View>;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
});
