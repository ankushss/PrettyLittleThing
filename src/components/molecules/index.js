export * from './DressItem';
export * from './CartItem';
