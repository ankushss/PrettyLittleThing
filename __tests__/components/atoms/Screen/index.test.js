import React from 'react';
import {create} from 'react-test-renderer';
import {Screen} from '../../../../src/components/atoms';

describe('Screen', () => {
  it('matches snapshot', () => {
    const component = create(<Screen />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
