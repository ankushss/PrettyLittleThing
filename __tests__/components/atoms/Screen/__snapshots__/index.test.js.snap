// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "padding": 20,
      },
      undefined,
    ]
  }
/>
`;
